sensor/common/deduper/deduper_test.go

```diff
@@ -81,3 +83,81 @@ func Test_DeduperParseKeyFromEvent(t *testing.T) {
 	assert.Equal(t, syncMessage.UnchangedIds[0], "Deployment:1234")
 
 }
+
+func Test_DeduperShallNotDedupeSomeMessages(t *testing.T) {
+	cases := map[string]struct {
+		msg        *central.MsgFromSensor
+		key        string
+		wantDedupe bool
+	}{
+		"Identical IndexReports shall not be deduped": {
+			msg: &central.MsgFromSensor{
+				Msg: &central.MsgFromSensor_Event{
+					Event: &central.SensorEvent{
+						Id: "1",
+						Resource: &central.SensorEvent_IndexReport{
+							IndexReport: &v4.IndexReport{
+								HashId:   "nodeID",
+								State:    "7", // IndexFinished
+								Success:  true,
+								Err:      "",
+								Contents: &v4.Contents{},
+							},
+						},
+					},
+				},
+			},
+			key:        "IndexReport:1",
+			wantDedupe: false,
+		},
+		"Identical ServiceAccounts shall be deduped": { // as an example of something that should be deduped
+			msg: &central.MsgFromSensor{
+				Msg: &central.MsgFromSensor_Event{
+					Event: &central.SensorEvent{
+						Id: "1",
+						Resource: &central.SensorEvent_ServiceAccount{
+							ServiceAccount: &storage.ServiceAccount{
+								Id:               "1",
+								Name:             "abc",
+								Namespace:        "ns",
+								ClusterName:      "cluster1",
+								ClusterId:        "0abcdef",
+								Labels:           nil,
+								Annotations:      nil,
+								CreatedAt:        nil,
+								AutomountToken:   false,
+								Secrets:          nil,
+								ImagePullSecrets: nil,
+							},
+						},
+					},
+				},
+			},
+			key:        "ServiceAccount:1",
+			wantDedupe: true,
```

Comment:
I thought we had tests for the events that need to be deduped 🤔 If we don't, I think we should add them. Also for other events that should not be deduped like runtime events.

Reply:
I could not find them in the package. I can add cases for the two or three other events that should not be deduped.

Reply:
Here it is: https://github.com/stackrox/stackrox/pull/13633